### PR TITLE
Fix all the pandas warnings

### LINKED ISSUE #211: replace deprecated pandas.Float64Index
From the tests
```
src/alchemlyb/tests/parsing/test_gmx.py: 61 warnings
  /home/runner/work/alchemlyb/alchemlyb/src/alchemlyb/parsing/gmx.py:210: FutureWarning: pandas.Float64Index is deprecated and will be removed from pandas in a future version. Use pandas.Index with the appropriate dtype instead.
    index=pd.Float64Index(times.values, name='time'))
```

### PR DESCRIPTION
**Pull request by xiki-tempula**
Fix #211

### CODE REVIEW

File: src/alchemlyb/preprocessing/subsampling.py [hunk 1]
```diff
@@ -130,13 +130,13 @@ def decorrelate_dhdl(df, drop_duplicates=True, sort=True, **kwargs):
...
-        return df.sort_index(0).reset_index('time', name='').duplicated('time').any()
+        return df.sort_index(axis=0).reset_index('time', name='').duplicated('time').any()
     else:
-        return df.sort_index(0).reset_index('time').duplicated('time').any()
+        return df.sort_index(axis=0).reset_index('time').duplicated('time').any()
 
 
 def _check_sorted(df):
-    return df.reset_index(0)['time'].is_monotonic_increasing
+    return df.reset_index(axis=0)['time'].is_monotonic_increasing
```
Comment (orbeckst) [inline src/alchemlyb/preprocessing/subsampling.py:139]: Tests fail with *TypeError: reset_index() got an unexpected keyword argument 'axis'*

**Review by orbeckst — CHANGES_REQUESTED** [review]
Test failures due to reset_indx(axis=...), please fix

**Comment by xiki-tempula** [comment]
@orbeckst Sorry, I think I might have `find & replace` something that I shouldn't.

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
thanks!!

**Pull request merged**